Seneca-CDOT/telescope
# Fix #746, #760, #946: enhance the My Feeds page

## Issue This PR Addresses



Fixes #746
Fixes #760
Fixes #946

## Type of Change




## Description



Fixes #746 by finalizing the implementation of buttons on the My Feeds page:
- Clicking on the `?` button of the My Feeds page now opens a [popover](https://material-ui.com/api/popover/) containing a helpful message.
   - Uses [`material-ui-popup-state`](https://www.npmjs.com/package/material-ui-popup-state) (simplifies popover state management)
- Introduces form handling/validation via [`react-material-ui-form-validator`](https://www.npmjs.com/package/react-material-ui-form-validator).
   - (One of the custom-defined validation rules uses the `valid-url` module, which was previously added to Telescope for use in our backend.)
- Calling `addFeed()` now causes a status message to be rendered to the page, indicating whether its fetch request to add a feed succeeded or failed.
- When the page is loaded, each of the current user's owned feeds are fetched and rendered to the page within separate (disabled) text boxes.

Fixes #760 by preventing access to the page by users who are not logged in.
- See: https://github.com/Seneca-CDOT/telescope/issues/760#issuecomment-610695732

Fixes #946 by adding feed deletion buttons alongside each existing feed.
   - **_Update: these now send a `DELETE /feeds/:id` request when clicked._**

_How to review:_
1. Run `docker-compose up` (ensure that `API_URL=http://localhost:3000` in `.env`)
1. Run `npm run clean; npm run build` to build the Gatsby frontend
1. Log into Telescope (http://localhost:3000/auth/login, user1/user1pass)
1. Navigate to the My Feeds page (http://localhost:3000/myfeeds)
1. Fill out the two form controls and click the '+' button to add a new feed. Doing so should cause the new feed to (after a few seconds) appear as a new set of form controls.
1. Clicking on the '?' button should pop up an informative message to assist with the previous step.
1. Wait several seconds after the page refreshes: the new feed should appear within an additional set of (disabled) form controls.
1. Clicking on the 'delete' icon (next to the set of disabled controls) should spawn a dialog window confirming the action. Confirming this dialog will send a `DELETE /feeds/:id` request (which won't result in anything happening until the corresponding backend functionality is merged and this PR is rebased over it).

## Checklist
